### ISSUE TITLE
Linux Mint 18.3 - Error:Couldn't launch the stream. Unexpected version check output [SOLVED]

### ISSUE THREAD

**Issue by blndr-tommy1441** [streamlink/streamlink-twitch-gui#636]
## Issue

- [X] This is not a bug report, feature request, or plugin issue/request.
- [X] I have read the contribution guidelines.

### Description

First I would like to apologize if this is not the right way or place to report an issue. This is my first time actually doing so.

I have Linux Mint 18.3 Cinnamon 64bit installed on my main PC (Kernel 4.15.0-47-generic)
I very recently updated Streamlink Twitch GUI to 1.7.1 and now I cannot open any stream. I get the following error

"Error:Couldn't launch the stream. Unexpected version check output"

Honestly I don't know if it's a GUI issue, a Streamlink issue or a Python issue. I don't know where to start to troubleshoot as I am fairly new to Linux environment. 

FYI, I also run both Streamlink and Streamlink Twitch GUI on a Windows 10 laptop and they were all updated without any problems and run perfectly. So my guess is a Linux issue.

And for both the PC and the laptop I use VLC Media Player.

So any pointers as how to resolve the issue will be greatly appreciated and if it could be ELI5 even better.

Thanks in advance.

### Reproduction steps / Explicit stream URLs to test

1. ...
2. ...
3. ...

### Logs

```
REPLACE THIS TEXT WITH THE LOG OUTPUT
```

### Additional comments, screenshots, etc.

![Screenshot from 2019-04-23 22-26-16](https://user-images.githubusercontent.com/19252685/56628189-daf03600-6616-11e9-817b-eeaae2fd6fdd.png)
![Screenshot from 2019-04-23 22-25-13](https://user-images.githubusercontent.com/19252685/56628162-bf852b00-6616-11e9-90d2-6a1c21ab5441.png)

![Screenshot from 2019-04-23 22-27-32](https://user-images.githubusercontent.com/19252685/56628238-0d019800-6617-11e9-8a26-a77c9f445ab1.png)
**Note:** I've tried changing to run on either python2.7 or  python3 and the error still occurs

![Screenshot from 2019-04-23 22-31-33](https://user-images.githubusercontent.com/19252685/56628485-fc055680-6617-11e9-8d47-04cc179e3f3d.png)

[Love Streamlink? Please consider supporting our collective. Thanks!](https://opencollective.com/streamlink/donate)

**Comment by bastimeyer** [comment]
The Unexpected version check error simply means that the Twitch GUI wasn't able to initially check Streamlink's version string because it returned something else, eg. an error message. In this case, you can try running Streamlink from the command line shell and see whether it's a Streamlink error or a config error in the Twitch GUI (eg. when chossing a wrong python interpreter).

If you remove the custom paths, the Twitch GUI should be able to automatically choose the right config - unless Streamlink itself is broken on your system. I don't know exactly how Streamlink is packaged on Mint, but it's possible that it requires python 3.

If this doesn't work, run this and paste the output of this into the respecitive fields
```bash
# Streamlink path
which streamlink
# Python path
cat $(which streamlink) | head -n1 | cut -c3-
```

**Comment by blndr-tommy1441** [comment]
Thanks @bastimeyer for your prompt reply,

I ran streamlink in the CLI and this was the output:
![Screenshot from 2019-04-23 23-27-51](https://user-images.githubusercontent.com/19252685/56630548-ba78a980-661f-11e9-9bc0-1b1cdf6a8400.png)

Then tried removing the custom paths and still same error. And finally ran the commands and pasted the outputs into the respective fields and nothing, I still get the error.

Now is there a way to uninstall completely both Streamlink and Streamlink GUI and start from scratch???

**Comment by bastimeyer** [comment]
You need to fix your Streamlink install. Your error is completely unrelated to Streamlink Twitch GUI. How did you install Streamlink on Mint?

Just follow Streamlink's install documentation
https://streamlink.github.io/install.html

**Comment by blndr-tommy1441** [comment]
Ok found a way to completely uninstall Streamlink from the PC...rebooted the PC.
Followed the Linux Ubuntu steps to install Streamlink, when I checked version it gave me the same errors as before.
Just for the sake of ticking all the boxes, checked python2.7 and python3 if there were some issues or dependencies that could be causing the problem...no issues.

**Comment by bastimeyer** [comment]
> I have Linux Mint 18.3 Cinnamon 64bit
> Followed the Linux Ubuntu steps

https://launchpad.net/~nilarimogard/+archive/ubuntu/webupd8/+index?field.series_filter=xenial
You can't use the `webupd8` PPA then, as you will get version `0.9.0-1~webupd8~xenial3` (Mint 18.3 is based on Ubuntu 16.04 "Xenial Xerus"), which doesn't meet the requirements when running the latest version of the Twitch GUI. The PPA works fine though, so you must have messed something up on your system.

In order to get the latest Streamlink version, try to install it via `pip`/`pip3` (`python-pip`/`python3-pip` packages on Ubuntu based distros):
https://streamlink.github.io/install.html#source-code

**Comment by blndr-tommy1441** [comment]
Ok I'm headed in the right direction.

- Uninstalled Streamlink from the PC, rebooted
- Installed it via the Latest pip and it worked fine, just a new issue developed

It runs fine via CLI and it shows the stream with the exception of requests dependency warning

![Screenshot from 2019-04-24 09-29-23](https://user-images.githubusercontent.com/19252685/56666207-74503400-6679-11e9-9466-9083c87f3d36.png)

But when I run the GUI I get the same issue.

I've changed the settings, leaving it blank, pointed to either python2.7 or python3, I made sure that the link for Streamlink is pointed towards the working install and tried every possible permutation that I could think of and still get the "Unexpected version check output" error

I apologize for being such a pain and wish to extend my full gratitude in helping me sort this one out

**Comment by bastimeyer** [comment]
> It runs fine via CLI

No, it does not, because it writes a warning to `stdout` and thus makes the Twitch GUI fail to initially parse its version number.

You were using Python 2.7 while installing Streamlink via pip and there seems to be a dependency issue with `liburl3`, same issue as
https://github.com/streamlink/streamlink/issues/2446

I have tested this in a fresh Mint 18.3 VM and installing Streamlink via python3/pip3 is working just fine:
```bash
$ sudo apt install python3-pip python3-setuptools
...
$ pip3 install --user streamlink
... # some build errors, read below
$ which streamlink
/home/mint/.local/bin/streamlink
$ head -n1 $(which streamlink) | cut -c3-
/usr/bin/python3
$ streamlink --version
streamlink 1.1.1
```

These Ubuntu versions which Mint is based on are still using the deprecated Python 2.7 by default whose support will be running out this year after an extended time of almost 10 years. Make sure that you're using Python3 when installing Streamlink. You will probably also have to remove the previously installed dependencies and pip cache.

You can also use `pip3` to install the latest `pip` (while using python3). This was necessary for me to fix the dependency build issues, as Mint's packaged pip2/3 versions were very much out of date. I'm not sure though if you will need this.
```bash
$ sudo apt install python3-setuptools python3-pip
...
$ pip3 install --user pip
...
$ pip --version
pip 19.1 from /home/mint/.local/lib/python3.5/site-packages/pip (python 3.5)
$ pip install --user streamlink
... # no build errors
$ streamlink --version
streamlink 1.1.1
```

**Comment by blndr-tommy1441** [comment]
> No, it does not, because it writes a warning to `stdout` and thus makes the Twitch GUI fail to initially parse its version number.

My bad when I said that it was working, it opened VLC and showed the stream with no lag or issues. Apologies

Anyway, I followed your first steps installing using pthon3/pip3 and now is 100% running...it opens VLC player without a hitch or hiccups.

So marking out the issue **[Solved]**.

Thanks for your patience and help!

**Comment by mcvons** [comment]
quick related query: best way to uninstall ?? (sorry for such a newb query)
